packages/browser/src/plugins/segmentio/batched-dispatcher.ts

```diff
@@ -107,6 +138,7 @@ export default function batch(
     if (pageUnloaded && buffer.length) {
       const reqs = chunks(buffer).map(sendBatch)
       Promise.all(reqs).catch(console.error)
+      // can we handle a retry here?
```

Comment:
Unfortunately not -- there are very serious restrictions around what you can do onPageChange. Sending an HTTP request to the server when the user closes the browser can be a bit tricky, as there's no guaranteed way to ensure the request will be completed successfully before the browser closes. This is one of the more brittle parts of the application.